---
title: "Deseq2"
author: "Linnea Honeker"
date: "4/6/2022"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("DESeq2")
packageVersion("DESeq2")
library(dplyr)
library(knitr)
library(apeglm)
library("ggplot2")
library("pheatmap")
library("stringr")
library(tidyverse)
library(cowplot)
library(reshape2)
```

# WGCNA analysis of soil pyruvate metatranscriptomic data. 

```{r load_data}
#import count data, transform to matrix, import column metadata, check that its a data.frame
cts.0 <- read.csv("../Data/Deseq2/Drought_vs_predrought_metaT_missing_1_no_feature.csv", header = TRUE, sep =",")
coldata <- read.csv("../Data/Deseq2/metaT-metadata.csv", header = TRUE, sep = ",")
cts.02 <- as.matrix(cts.0)
row.names(cts.02) <- cts.02[,1]
cts.02 <- cts.02[,-1]

cts <- matrix(as.numeric(cts.02),    # Convert to numeric matrix
                  ncol = ncol(cts.02))
cts.0$Feature = NULL
colnames(cts) <- colnames(cts.0)

row.names(cts) <- row.names(cts.02)

row.names(coldata) <- coldata$SampleID
coldata$SampleID <- NULL
```

## Drought vs. Pre-drought across all timepoints post pyruvate addition (0, 6, and 48 hr)
### Step 1
Create deseq2 object to compare drought vs pre-drought across all time points

```{r create_deseq2_object_drought_vs_predrought}

###create a deseq2 object with Condition, Time, and Site as design###
coldata$Condition <- factor(coldata$Condition, c("PreDrought", "Drought"))

coldata$Condition <- relevel(coldata$Condition, ref = "PreDrought")


dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Condition + Site)
```

### Step 2
Explore data using variance stabilization normalization (VST) transformation

```{r data_exploration}
vstcounts <- vst(dds, blind=TRUE)
vst.export<- assay(vstcounts)
vst.export<- as.data.frame(vst.export)
write.csv(vst.export, "../Output/Deseq2/vst-metaT.csv")

plotPCA(vstcounts, intgroup=c("Condition"))

select <- order(rowMeans(counts(dds)),
                decreasing=TRUE)[1:500]

df <- as.data.frame(colData(dds)[,c("Condition","Site")])

pheatmap(assay(vstcounts)[select,], cluster_rows=TRUE, show_rownames=TRUE,
         cluster_cols=TRUE, annotation_col=df)

```

### Step 3
Run Deseq2 on object to find differential gene expression. Here we are comparing drought vs. pre-drought across all time points

```{r DE_analysis_drought_vs_predrought}
dds <- DESeq(dds)
resultsNames(dds) # lists the coefficients

##drought vs predrought time 0
res <- results(dds, contrast = c("Condition","Drought","PreDrought"))
plotMA(res, ylim=c(-40,40))

summary(res)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-10,30), ylim=c(0,20)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

Shrink effect size using apeglm. Notice spread of log2FC decreases without the extreme outliers of log2FC > 20. 

```{r}
#shrink effect size
resLFC <- lfcShrink(dds, coef="Condition_Drought_vs_PreDrought", type="apeglm")
resLFC
plotMA(resLFC, ylim=c(-2,2))

summary(resLFC)

#make volcano plot
#reset par
par(mfrow=c(1,1))

# Make a basic volcano plot
with(resLFC, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-11,10), ylim=c(0,22)))
with(subset(resLFC, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLFC, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r export_results_drought_vs_predrought, include=FALSE}
#export results
write.csv(as.data.frame(res), 
          file="../Output/Deseq2/drought-vs-predrought.csv")
write.csv(as.data.frame(resLFC),
          file="../Output/Deseq2/drought-vs-predrought-lfc.csv")
```

## Drought vs. Pre-drought across different time points. Here, pre-drought 0hr is set as reference.)
### Step 1
Create deseq2 object 

```{r create_deseq2_object_ref_predrought0}
###create a deseq2 object with Condition, Time, and Site as design###
coldata$Condition_Time <- factor(coldata$Condition_Time, c("PreDrought_0hr", "PreDrought_6hr", "PreDrought_48hr",
                                                           "Drought_0hr", "Drought_6hr", "Drought_48hr"))

coldata$Condition_Time <- relevel(coldata$Condition_Time, ref = "PreDrought_0hr")


dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Condition_Time + Site)

```

### Step 2
Explore data using variance stabilization normalization (VST) transformation

```{r data_exploration_drought_vs_predrought}

vstcounts <- vst(dds, blind=TRUE)
vst.export<- assay(vstcounts)
vst.export<- as.data.frame(vst.export)

plotPCA(vstcounts, intgroup=c("Condition_Time"))

select <- order(rowMeans(counts(dds)),
                decreasing=TRUE)[1:500]
df <- as.data.frame(colData(dds)[,c("Condition_Time","Site")])


pheatmap(assay(vstcounts)[select,], cluster_rows=TRUE, show_rownames=TRUE,
         cluster_cols=TRUE, annotation_col=df)

```

### Step 3a
Run Deseq2 on object to find differential gene expression. Here we are comparing Drought 0hr vs Pre-drought 0hr

```{r DE_analysis_drought_0hr_vs_predrought_0hr}
###deifferential expression analysis###

dds <- DESeq(dds)
resultsNames(dds) # lists the coefficients

##drought vs predrought time 0
res.T0 <- results(dds, contrast = c("Condition_Time","Drought_0hr","PreDrought_0hr"))
plotMA(res, ylim=c(-40,40))

summary(res.T0)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res.T0, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-40,40), ylim=c(0,40)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res.T0, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res.T0, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))


```

### Step 3b
Shrink effect size using apeglm. Notice spread of log2FC decreases without the extreme outliers of log2FC > 20. 

```{r}
#shrink effect size
resLFC.T0 <- lfcShrink(dds, coef="Condition_Time_Drought_0hr_vs_PreDrought_0hr", type="apeglm")
resLFC.T0
plotMA(resLFC.T0, ylim=c(-2,2))

summary(resLFC.T0)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(resLFC.T0, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-15,10), ylim=c(0,25)))
with(subset(resLFC.T0, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLFC.T0, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r export_results_drought_0hr_vs_predrought_0hr, include=FALSE}
write.csv(as.data.frame(res.T0), 
          file="../Output/Deseq2/drought-0hr-vs-predrought-0hr.csv")
write.csv(as.data.frame(resLFC.T0),
          file="../Output/Deseq2/drought-0hr-vs-predrought-0hr-lfc.csv")
```

## Drought vs. Pre-drought across different time points. (Here, pre-drought 6hr is set as reference.)
### Step 1
Create deseq2 object 
 
```{r}
###create a deseq2 object with Condition, Time, and Site as design###
coldata$Condition_Time <- factor(coldata$Condition_Time, c("PreDrought_0hr", "PreDrought_6hr", "PreDrought_48hr",
                                                           "Drought_0hr", "Drought_6hr", "Drought_48hr"))

coldata$Condition_Time <- relevel(coldata$Condition_Time, ref = "PreDrought_6hr")


dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Condition_Time + Site)
```


### Step 3a (Skip step 2, data exploration, because already performed above)
Run Deseq2 on object to find differential gene expression. Here we are comparing Drought 0hr vs Pre-drought 0hr

```{r DE_analysis_of_drought_6hr_vs_predrought_6hr}
dds <- DESeq(dds)
resultsNames(dds) # lists the coefficients

res.T6 <- results(dds, contrast = c("Condition_Time","Drought_6hr","PreDrought_6hr"))
plotMA(res.T6, ylim=c(-40,40))

summary(res.T6)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res.T6, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-40,40), ylim=c(0,40)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res.T6, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res.T6, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

### Step 3b
Shrink effect size using apeglm. Notice spread of log2FC decreases without the extreme outliers of log2FC > 20. 

```{r}
#shrink effect size
resLFC.T6 <- lfcShrink(dds, coef="Condition_Time_Drought_6hr_vs_PreDrought_6hr", type="apeglm")
resLFC.T6
plotMA(resLFC.T6, ylim=c(-2,2))

summary(resLFC.T6)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(resLFC.T6, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-15,10), ylim=c(0,25)))
with(subset(resLFC.T6, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLFC.T6, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


```{r export_results_drought_6hr_vs_predrought_6hr, include = FALSE}
#export results
write.csv(as.data.frame(res.T6), 
          file="../Output/Deseq2/drought-6hr-vs-predrought-6hr.csv")
write.csv(as.data.frame(resLFC.T6),
          file="../Output/Deseq2/drought-6hr-vs-predrought-6hr-lfc.csv")
```

## Drought vs. Pre-drought across different time points. (Here, pre-drought 48hr is set as reference.)
### Step 1
Create deseq2 object 

```{r}
coldata$Condition_Time <- factor(coldata$Condition_Time, c("PreDrought_0hr", "PreDrought_6hr", "PreDrought_48hr",
                                                    "Drought_0hr", "Drought_6hr", "Drought_48hr"))

coldata$Condition_Time <- relevel(coldata$Condition_Time, ref = "PreDrought_48hr")


dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Condition_Time + Site)
```

### Step 3a (Skip step 2, data exploration, because already performed above)
Run Deseq2 on object to find differential gene expression. Here we are comparing Drought 0hr vs Pre-drought 0hr

```{r DE_analysis_of_drought_48hr_vs_predrought_48hr}
###deifferential expression analysis###
dds <- DESeq(dds)
resultsNames(dds) # lists the coefficients

res.t48 <- results(dds, contrast = c("Condition_Time","Drought_48hr","PreDrought_48hr"))
plotMA(res, ylim=c(-40,40))

summary(res)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res.t48, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-40,40), ylim=c(0,40)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res.t48, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res.t48, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))

```

### Step 3b
Shrink effect size using apeglm. Notice spread of log2FC decreases without the extreme outliers of log2FC > 20. 

```{r}
resLFC.t48 <- lfcShrink(dds, coef="Condition_Time_Drought_48hr_vs_PreDrought_48hr", type="apeglm")
resLFC.t48
plotMA(resLFC, ylim=c(-2,2))

summary(resLFC.t48)

#make volcano plot
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(resLFC.t48, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-15,10), ylim=c(0,25)))
with(subset(resLFC.t48, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(resLFC.t48, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```
```{r export_results_drought_48hr_vs_predrought_48hr, include = FALSE}
#export results
write.csv(as.data.frame(res.t48), 
          file="../Output/Deseq2/drought-48hr-vs-predrought-48hr.csv")
write.csv(as.data.frame(resLFC.t48),
          file="../Output/Deseq2/drought-48hr-vs-predrought-48hr-lfc.csv")
```

# Notes on observations
## Most DE occurs at T=0, before pyruvate addition. During this time, the stress difference between pre-drought and drought conditions are the greatest.
## The least DE occurs at T=6, just after pyruvate addition. This must stimulate similar microbial activity, of perhaps those that lost activity during drought, than that of pre-drought.  
## LFC tends to cause the really low Log2FC values that I have been noticing. I think using the analysis of non-LFC transformed data is best

# Next, find DE of target genes involved in VOC cycling
## All time points
```{r}
#Extract acetate, acetone, and diacetyl genes of interest. Also, combine interest table with correlations of VOCs and gene expression from WGNCA analysis
interest <- read.csv("../Data/Deseq2/KO_interest.csv", header = TRUE)
correlations_acetone <- read.csv("../Data/Deseq2/gene-screening-acetone-spearman.csv", header = TRUE)
correlations_acetate <- read.csv("../Data/Deseq2/gene-screening-acetatec2-spearman.csv", header = TRUE)
correlations_diacetyl <- read.csv("../Data/Deseq2/gene-screening-diacetyl-spearman.csv", header = TRUE)

#create a correlation matrix between each VOC and genes of interest. Create a separate table for q values
corr_table <- interest %>%
  merge(correlations_acetone, by = "KO") %>%
  rename(Acetone = cor.Standard) %>%
  select(c("gene_name", "Acetone", "KO", "VOC")) %>%
  merge(correlations_acetate, by = "KO") %>%
  rename(Acetate = cor.Standard) %>%
  select(c("gene_name", "Acetone", "Acetate", "KO", "VOC")) %>%
  merge(correlations_diacetyl, by = "KO") %>%
  rename(Diacetyl = cor.Standard) %>%
  arrange(VOC) %>%
  select(c("gene_name", "Acetone", "Acetate", "Diacetyl")) %>%
  pivot_longer(!gene_name, names_to = "VOC", values_to = "corr")

#q_table <-   
#heatmap
ggplot(data = corr_table, aes(x=gene_name, y= VOC, fill = corr)) +
  geom_tile()

##all times
as.data.frame(res) %>%
  rownames_to_column(var = "KO") %>%
  merge(interest , by = "KO") %>%
  #filter(Priority == "yes") %>%
  mutate(Significance =
           case_when(padj < 0.05 & log2FoldChange > 0 ~ "Sig.D", 
                     padj < 0.05 & log2FoldChange < 0 ~ "Sig.PD")) %>%
  mutate(datExpr = str_replace(KO, 'KO:', '')) %>%
  mutate(gene_name = fct_reorder(gene_name, log2FoldChange)) %>%
  ggplot(aes(x=gene_name, y=log2FoldChange, fill = Activity, color = Significance)) +
  geom_col() +
  ylim(-6, 6) +
  coord_flip() +
  theme_bw() +
  labs(y="Log-2FC") +
  facet_wrap(~VOC, scales = "free", ncol = 1) +
  scale_fill_manual (values = c("lightgray","black"),
                     breaks = c("consume", "produce")) +
  scale_color_manual (values = c("#2ECC71" ,"#F39C12"),
                      breaks = c("Sig.PD", "Sig.D")) +
  theme(text = element_text(size = 12,
                            family = "Arial",
                            color = "black"),
        legend.title = element_blank() ,
        legend.position = "bottom",
        axis.title.y = element_blank(),
        axis.title.x = element_text(face= "bold"), 
        axis.text.x = element_text(vjust = 1, hjust = 0.5, color = "black", size = 10),
        axis.text.y = element_text(color = "black", size = 10))
ggsave("../Figures/Deseq2/VOC-genes-all-times.png", dpi = 300, width = 3, height = 5)



```




###Time 0 hr

```{r}
#time 0
p1 <- as.data.frame(res.T0) %>%
  rownames_to_column(var = "KO") %>%
  merge(.,interest , by = "KO") %>%
  #filter(Priority == "yes") %>%
  mutate(Significance =
           case_when(padj < 0.05 & log2FoldChange > 0 ~ "Sig.D", 
                     padj < 0.05 & log2FoldChange < 0 ~ "Sig.PD")) %>%
  mutate(datExpr = str_replace(KO, 'KO:', '')) %>%
  mutate(gene_name = fct_reorder(gene_name, log2FoldChange)) %>%
  ggplot(aes(x=gene_name, y=log2FoldChange, fill = Activity, color = Significance)) +
  geom_col() +
  coord_flip() +
  theme_bw() +
  labs(y="Log-2FC") +
  ylim(-30,20) +
  facet_wrap(~VOC, scales = "free", ncol = 1) +
  scale_fill_manual (values = c("lightgray","black"),
                     breaks = c("consume", "produce")) +
  scale_color_manual (values = c("#2ECC71" ,"#F39C12"),
                      breaks = c("Sig.PD", "Sig.D")) +
  theme(text = element_text(size = 12,
                            family = "Arial",
                            color = "black"),
        legend.title = element_blank() ,
        legend.position = "bottom",
        axis.title.y = element_blank(),
        axis.title.x = element_text(face= "bold"), 
        axis.text.x = element_text(vjust = 1, hjust = 0.5, color = "black", size = 10),
        axis.text.y = element_text(color = "black", size = 10))
ggsave("../Figures/Deseq2/VOC-genes-T0.png", dpi = 300, width = 3, height = 5)

# return vecor with correct gene order from plot above
ordered.res.T0 <- as.data.frame(res.T0) %>%
   rownames_to_column(var = "KO") %>%
  merge(.,interest , by = "KO") %>%
  #filter(Priority == "yes") %>%
  arrange(log2FoldChange)

ordered.VOC.vector <- ordered.res.T0$gene_name

```
###Time 6 hr
```{r}
#time 6
p2 <- as.data.frame(res.T6) %>%
  rownames_to_column(var = "KO") %>%
  merge(.,interest , by = "KO") %>%
  #filter(Priority == "yes") %>%
  mutate(Significance =
           case_when(padj < 0.1 & log2FoldChange > 0 ~ "Sig.D", 
                     padj < 0.1 & log2FoldChange < 0 ~ "Sig.PD")) %>%
  mutate(datExpr = str_replace(KO, 'KO:', '')) %>%
  mutate(gene_name = factor(gene_name, ordered.VOC.vector)) %>%
  ggplot(aes(x=gene_name, y=log2FoldChange, fill = Activity, color = Significance)) +
  geom_col() +
  coord_flip() +
  theme_bw() +
  labs(y="Log-2FC") +
  ylim(-8,8) +
  facet_wrap(~VOC,  scales = "free", ncol = 1) +
  scale_fill_manual (values = c("lightgray","black"),
                     breaks = c("consume", "produce")) +
  scale_color_manual (values = c("#2ECC71" ,"#F39C12"),
                      breaks = c("Sig.PD", "Sig.D")) +
  theme(text = element_text(size = 12,
                            family = "Arial",
                            color = "black"),
        legend.title = element_blank() ,
        legend.position = "bottom",
        axis.title.y = element_blank(),
        axis.title.x = element_text(face= "bold"), 
        axis.text.x = element_text(vjust = 1, hjust = 0.5, color = "black", size = 10),
        axis.text.y = element_text(color = "black", size = 10))
ggsave("../Figures/Deseq2/VOC-genes-T6.png", dpi = 300, width = 3, height = 5)
```
###Time 48 hr
```{r}
#time 48
p3 <- as.data.frame(res.t48) %>%
  rownames_to_column(var = "KO") %>%
  merge(.,interest , by = "KO") %>%
  #filter(Priority == "yes") %>%
  mutate(Significance =
           case_when(padj < 0.1 & log2FoldChange > 0 ~ "Sig.D", 
                     padj < 0.1 & log2FoldChange < 0 ~ "Sig.PD")) %>%
  mutate(datExpr = str_replace(KO, 'KO:', '')) %>%
  mutate(gene_name = factor(gene_name, ordered.VOC.vector)) %>%
  ggplot(aes(x=gene_name, y=log2FoldChange, fill = Activity, color = Significance)) +
  geom_col() +
  coord_flip() +
  theme_bw() +
  labs(y="Log-2FC") +
  ylim(-8,8) +
  facet_wrap(~VOC, scales = "free", ncol = 1) +
  scale_fill_manual (values = c("lightgray","black"),
                     breaks = c("consume", "produce")) +
  scale_color_manual (values = c("#2ECC71" ,"#F39C12"),
                      breaks = c("Sig.PD", "Sig.D")) +
  theme(text = element_text(size = 12,
                            family = "Arial",
                            color = "black"),
        legend.title = element_blank() ,
        legend.position = "bottom",
        axis.title.y = element_blank(),
        axis.title.x = element_text(face= "bold"), 
        axis.text.x = element_text(vjust = 1, hjust = 0.5, color = "black", size = 10),
        axis.text.y = element_text(color = "black", size = 10))
ggsave("../Figures/Deseq2/VOC-genes-T48.png", dpi = 300, width = 3, height = 5)
```

```{r fig, fig.height = 6, fig.width = 6}
plot_grid(p1, p2, p3, 
          labels = c('A', 'B', 'C'),
          label_size = 12,
              ncol = 3)

```


